lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -296,6 +297,7 @@ sub engine_workit {
     my $assetkeys = detect_asset_keys(\%vars);
     if (my $cache_dir = $global_settings->{CACHEDIRECTORY}) {
         $shared_cache = do_asset_caching($job, \%vars, $cache_dir, $assetkeys, $webui_host, $pooldir);
+        return $shared_cache if (ref($shared_cache) eq 'HASH');
```

Comment:
seems from my original proposal you turned this around. Can you now check for a correct value rather than check the ref type? I don't think checking perl types is a good style here.

Reply:
hm, we might not have a better choice here as we can't check something like `if $shared_cache->{error}` if `$shared_cache` is actually a string. I think the current way as you provided is fine